PhoneBook.cpp

```diff
@@ -0,0 +1,55 @@
+#include "PhoneBook.h"
+
+PhoneBook::PhoneBook() :
+numberOfPages(1000),
+bookWeight(2.2f),
+telephoneCompany("Yellowpages"),
+bookAge(3.1f),
+glueStrength(1.7f)
+{
+    std::cout << "Delivering new Phonebook!" << std::endl;
+}
+
+PhoneBook::~PhoneBook()
+{
+    std::cout << "The Phonebook has been thrown away." << std::endl;
+}
+
+std::string PhoneBook::contactInformationDiplay(std::string contactName) const
+{
+    std::string phoneNumber = contactName + " 555-5555";
+    return phoneNumber;
+}
+
+void PhoneBook::openToRandomPage() const
+{
+    int min = 1;
+    int max = numberOfPages;
+    int randomNumber = rand() % (max - min + 1) + min;
+    std::cout << "Opening page, out of " << numberOfPages << " pages, we open to "<< randomNumber << std::endl;
+}
+
+void PhoneBook::disintigrate()
+{
+    std::cout << "Uahghghgllll the " << telephoneCompany << " phone book is disintigrating" << std::endl;
+    numberOfPages = 0;
+}
+
+int PhoneBook::tearOutPages()
+{
+    int pagesTorn = 0;
+    while (pagesTorn < numberOfPages)
+    {
+        int pagesToTear = (numberOfPages - pagesTorn) / 2 + 25;
+        if (pagesTorn + pagesToTear > numberOfPages) 
+        {
+            pagesTorn = numberOfPages; 
+        }
+        {
+            pagesTorn += pagesToTear;
+            std::cout << "Tearing out " << pagesToTear << " pages." << std::endl;
+        }
```

Comment:
Why are there extra curly braces here?
- [ ] revise

Reply:
The extra brackets removed. Also reconfigured the function as the "pagesToTear" variable was unnecessary and a bit weird. I'll be sure to reformulate old stuff as I go forward - initial function created with a misunderstanding of "if" syntax.